sound/soc/sof/ipc4-loader.c

```diff
@@ -256,26 +254,111 @@ static int sof_ipc4_load_library_by_uuid(struct snd_sof_dev *sdev,
 	if (ret)
 		goto release;
 
-	ret = xa_insert(&ipc4_data->fw_lib_xa, lib_id, fw_lib, GFP_KERNEL);
+	ret = xa_insert(&ipc4_data->fw_lib_xa, *lib_id, fw_lib, GFP_KERNEL);
 	if (unlikely(ret))
 		goto release;
 
-	kfree(fw_filename);
-
+	(*lib_id)++;
 	return 0;
 
 release:
 	release_firmware(fw_lib->sof_fw.fw);
 	/* Allocated within sof_ipc4_fw_parse_ext_man() */
 	devm_kfree(sdev->dev, fw_lib->modules);
-free_filename:
-	kfree(fw_filename);
 free_fw_lib:
 	devm_kfree(sdev->dev, fw_lib);
 
 	return ret;
 }
 
+/**
+ * sof_ipc4_load_library_bundles - loads the library parts of a modular firmware
+ * @sdev: SOF device
+ *
+ * With IPC4 the firmware can be monolithic or modular release.
```

Comment:
@ujfalusi you want to remove the monolithic reference in this PR too?

Reply:
yes, I wanted to... ;)